internal/version.go

```diff
@@ -20,13 +20,20 @@
 
 package internal
 
-// LibraryVersion is a semver string that represents
-// the version of this cadence client library
-// it will be embedded as a "version" header in every
-// rpc call made by this client to cadence server.
-// In addition, the version string will be used by
-// the server to enforce compatibility checks
-// Update to this version number is typically done
+// below are the metadata which will be embedded as part
+// of headers in every rpc call made by this client to
+// cadence server.
+
+// Update to the metadata below is typically done
 // by the cadence team as part of a major feature or
 // behavior change
+
+// LibraryVersion is a semver string that represents
+// the version of this cadence client library
 const LibraryVersion = "v0.5.0"
+
+// FeatureVersion is a int that represents the
+// feature set of this cadence client library support.
+// This can be used for client capibility check, for
+// backward compatibility
+const FeatureVersion = "1"
```

Comment:
I think it should also be semantic version with MAJOR.MINOR version of the server it was tested against. The server itself would have MAJOR.MINOR.PATCH version, where PATCH would be for any changes that are not API changing.

Reply:
so what about this:

internally, we keep 3 version number on the client side (major, minor, patch), and when exposing to customer who relying on the version number, we just exposed_version = sprintf("v%s.%s.%s"), i.e. "v1.2.3"

when piggy the version back to server, we send back 3 version separately, like:
"cadence-client-major-ver": "1"
"cadence-client-minor-ver": "2"
"cadence-client-patch-ver": "3"
to avoid parsing.

Reply:
I'm confused what version are you talking about?
For feature version only two numbers are necessary, and parsing is trivial. 
For client version there is no need for parsing yet.